### PR TITLE
Add function to support Alcor API Rally test

### PR DESCRIPTION
**Pull request by DavidLiu506**
Add function to support alcor api rally test:
Openstack network client does not support subnet routing rule update and delete which included in alcor.
Add these two api to support it.

### CODE REVIEW

File: neutronclient/v2_0/client.py [hunk 1]
```diff
@@ -967,12 +976,12 @@ def remove_interface_router(self, router, body=None):
 
     def add_extra_routes_to_router(self, router, body=None):
         """Adds extra routes to the specified router."""
-        return self.put((self.router_path % router) + "/add_extraroutes",
+        return self.put((self.router_path % router) + "/add_extra_routes",
```
Comment (cj-chung) [inline neutronclient/v2_0/client.py]: @DavidLiu506 please make a note here, the current `add_extra_routes` is not compatible with the latest neutron api. If we need to 100% compatible with latest neutron, we should modify our api to use `add_extraroutes`.

File: neutronclient/v2_0/client.py [hunk 2]
```diff
@@ -870,6 +870,15 @@ def list_subnets(self, retrieve_all=True, **_params):
         return self.list('subnets', self.subnets_path, retrieve_all,
                          **_params)
 
+    def update_subnet_routetable(self, subnet, body=None):
+        """Adds extra routes to the specified router."""
+        return self.put((self.subnet_path % subnet) + "/routetable",
+                        body=body)
```
Comment (cj-chung) [inline neutronclient/v2_0/client.py:876]: @DavidLiu506 Should we put another layer of API in rally's neutron.py? In that way, we can have a consistent way to call neutronclient's api from rally.

Comment (DavidLiu506) [reply neutronclient/v2_0/client.py:876]: Already follow rally test layer to do rally test. First write plugins to call rally (there two layers) and rally call this api talk to alcor. Same as rally-openstack.

Comment (DavidLiu506) [reply neutronclient/v2_0/client.py:876]: @cj-chung Thanks for your review. Forgot copy this lay to PR.

File: neutronclient/v2_0/client.py (hunk 1 above)
Comment (xieus) [reply neutronclient/v2_0/client.py]: Just curious about the difference about our current implementation and the latest Neutron add_extraroutes apis. If not much, let us make it compatible.

**Comment by xieus** [comment]
@DavidLiu506 A general feedback: could you please add description to a PR including this one and others going forward? In this way, we could always trace back to a PR later on. Thank you.

**Comment by DavidLiu506** [comment]
@xieus Thanks for your suggestion.

Will work on it.

File: neutronclient/v2_0/client.py (hunk 1 above)
Comment (cj-chung) [reply neutronclient/v2_0/client.py]: @DavidLiu506 Please make change in Route Manager for these APIs in order to compatible with latest Neutron version.

**Review by cj-chung — CHANGES_REQUESTED** [review]

**Review by cj-chung — APPROVED** [review]
LGTM

**Pull request merged**